File: TerraObserver/TO.GodotNodes.Abstractions/INode.cs
using Godot;

namespace TO.GodotNodes.Abstractions;

/// Copyright (C) 2025 Zhu Xiaohe(aka ZeromaXHe)
/// Author: Zhu XH
/// Date: 2025-04-16 16:49:16
public interface INode : IGodotObject
{
    event Action? Ready; // 2208
    event Action? TreeExiting; // 2252
    StringName Name { get; set; }
    NodePath GetPath();
    void SetProcess(bool enable);
    bool IsNodeReady();
    void AddChild(Node node, bool forceReadableName = false, Node.InternalMode @internal = (Node.InternalMode)(0));
    Node GetChild(int idx, bool includeInternal = false);
    Godot.Collections.Array<Node> GetChildren(bool includeInternal = false);
    Viewport GetViewport();
    void QueueFree();
}

File: TerraObserver/TO.Domains/Types/Godots/INode.fs
namespace TO.Domains.Types.Godots

open System
open System.Runtime.InteropServices
open Godot

/// Copyright (C) 2025 Zhu Xiaohe(aka ZeromaXHe)
/// Author: Zhu XH
/// Date: 2025-04-16 16:49:16
[<Interface>]
[<AllowNullLiteral>]
type INode =
    inherit IGodotObject
    abstract Name: StringName with get, set // 374

    abstract AddChild:
        node: Node *
        [<Optional; DefaultParameterValue(false)>] forceReadableName: bool *
        [<Optional; DefaultParameterValue( (*EnumOfValue<int64, Node.InternalMode> 0L*) Node.InternalMode.Disabled)>] ``internal``:
            Node.InternalMode ->
            unit // 791

    abstract GetChildCount: [<Optional; DefaultParameterValue(false)>] includeInternal: bool -> int // 827

    abstract GetChildren:
        [<Optional; DefaultParameterValue(false)>] includeInternal: bool -> Node Godot.Collections.Array // 839

    abstract GetChild: idx: int * [<Optional; DefaultParameterValue(false)>] includeInternal: bool -> Node // 861
    abstract GetPath: unit -> NodePath // 1066
    abstract SetProcess: bool -> unit // 1360
    abstract GetViewport: unit -> Viewport // 1873
    abstract QueueFree: unit -> unit // 1886
    abstract IsNodeReady: unit -> bool // 1910

    [<CLIEvent>]
    abstract Ready: Action IDelegateEvent // 2208

    [<CLIEvent>]
    abstract TreeExiting: Action IDelegateEvent // 2252
